clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -2794,12 +2794,12 @@ void
 raw2trace_t::unread_last_entry(raw2trace_thread_data_t *tdata)
 {
     VPRINT(5, "Unreading last entry\n");
+    tdata->pre_read.push_front(tdata->last_entry);
```

Comment:
Looks like it was PR #4994 that added the unread-split support: but the only test it adds is of the legacy kernel marker value which doesn't use this.  When is this unread hit?  Should be possible to add a unit test that exercises this, right?

Reply:
Existing control paths don't seem to hit this, so cannot add a unit test. An internal patch I'm testing needs this.

Reply:
> Existing control paths don't seem to hit this

Or atleast no control paths where the reversed order would manifest as a fatal error hit this.

Reply:
The reversed order showed up after the #7465 patch as a different raw2trace error "memref outside bb".

Reply:
Maybe a two-entry TRACE_MARKER_TYPE_KERNEL_EVENT marker would hit this for delayed branches where we have logic to unread_last_entry. Let me try writing a raw2trace_unit_test.

Reply:
I was able to reproduce the error using a large enough kernel_event marker value. This was never seen on a real trace probably because kernel_event marker values do not need the additional marker usually.

Reply:
(I should have added a large-kernel-event-marker unit test as part of PR #4994.)